selftests/unit/utils/software_manager.py

```diff
@@ -13,12 +13,26 @@ def apt_supported_distro():
     return distro.detect().name in ["debian", "Ubuntu"]
 
 
+def login_binary_path(distro_name, distro_version):
+    """Retrieve the login binary path based on the distro version""""
+    if distro_name == "Ubuntu":
+        if float(distro_version) >= 24.04:
+            return "/usr/bin/login"
+    if distro_name == "debian":
+        if distro_version == "trixie":
+            return "/usr/bin/login"
+    return "/bin/login"
+
+
 @unittest.skipUnless(os.getuid() == 0, "This test requires root privileges")
 @unittest.skipUnless(apt_supported_distro(), "Unsupported distro")
 class Apt(unittest.TestCase):
     def test_provides(self):
         sm = manager.SoftwareManager()
-        self.assertEqual(sm.provides("/bin/login"), "login")
+        distro_name = distro.detect().name
```

Comment:
And optimization here to avoid calling the detection twice:

```python
detected = distro.detect()
login_path = login_binary_path(detected.name, detected.version)
```

Also, it would be OK to change `login_binary_path` to take a `Distro` instance IMO.